universal-application-tool-0.0.1/test/services/applicant/ReadOnlyApplicantProgramServiceImplTest.java

```diff
@@ -8,60 +8,89 @@
+import support.ProgramBuilder;
+import support.TestQuestionBank;
 
 public class ReadOnlyApplicantProgramServiceImplTest extends WithPostgresContainer {
 
-  BlockDefinition blockA =
-      BlockDefinition.builder()
-          .setId(123L)
-          .setName("Block Name")
-          .setDescription("Block Description")
-          .build();
-
-  BlockDefinition blockB =
-      BlockDefinition.builder()
-          .setId(321L)
-          .setName("Block Name B")
-          .setDescription("Block Description B")
-          .build();
-
-  ProgramDefinition programDefinition =
-      ProgramDefinition.builder()
-          .setId(123L)
-          .setName("The Program")
-          .setDescription("This program is for testing.")
-          .addBlockDefinition(blockA)
-          .addBlockDefinition(blockB)
-          .build();
-
-  ReadOnlyApplicantProgramService subject;
+  private QuestionDefinition nameQuestion;
+  private QuestionDefinition colorQuestion;
+  private QuestionDefinition addressQuestion;
+  private ApplicantData applicantData;
+  private ProgramDefinition programDefinition;
+  private ReadOnlyApplicantProgramService subject;
 
   @Before
   public void setUp() {
-    subject =
-        new ReadOnlyApplicantProgramServiceImpl(
-            new Applicant().getApplicantData(), programDefinition);
+    applicantData = new ApplicantData();
+    nameQuestion = TestQuestionBank.applicantName().getQuestionDefinition();
+    colorQuestion = TestQuestionBank.applicantFavoriteColor().getQuestionDefinition();
+    addressQuestion = TestQuestionBank.applicantAddress().getQuestionDefinition();
+    programDefinition =
+        ProgramBuilder.newProgram()
+            .withBlock("Block one")
+            .withQuestionDefinition(nameQuestion)
+            .withBlock("Block two")
+            .withQuestionDefinition(colorQuestion)
+            .withQuestionDefinition(addressQuestion)
+            .buildDefinition();
+    subject = new ReadOnlyApplicantProgramServiceImpl(applicantData, programDefinition);
   }
 
   @Test
   public void getCurrentBlockList_getsTheApplicantSpecificBlocksForTheProgram() {
     ImmutableList<Block> blockList = subject.getCurrentBlockList();
 
-    assertThat(blockList.size()).isEqualTo(2);
+    assertThat(blockList).hasSize(2);
     Block block = blockList.get(0);
-    assertThat(block.getId()).isEqualTo(123L);
-    assertThat(block.getName()).isEqualTo("Block Name");
-    assertThat(block.getDescription()).isEqualTo("Block Description");
+    assertThat(block.getName()).isEqualTo("Block one");
+  }
+
+  @Test
+  public void getCurrentBlockList_doesNotIncludeCompleteBlocks() {
+    // Answer block one questions
+    answerNameQuestion();
+
+    ImmutableList<Block> blockList = subject.getCurrentBlockList();
+
+    assertThat(blockList).hasSize(1);
+    assertThat(blockList.get(0).getName()).isEqualTo("Block two");
+  }
+
+  @Test
+  public void getCurrentBlockList_returnsEmptyListIfAllBlocksComplete() {
```

Comment:
Question (blocking): shouldn't this return questions completed in this program?

Reply:
Great question - I updated the test name and comment to make it clearer that `getCurrentBlockList` should return empty if all blocks were completed in a different program.